client/state/selectors/is-connected-secondary-network-site.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { isMainSiteOf } from 'state/selectors';
+import { isJetpackSiteSecondaryNetworkSite } from 'state/sites/selectors';
+
+/**
+ * Returns true if site with id equal to siteId is a connected secondary network site and false otherwise
+ * Returns null if with the information available it's not possible to determine if site is a connected secondary site or not
+ * In order for a site to be a connected secondary site its main site also needs to be connected
+ * With selector isJetpackSiteSecondaryNetworkSite the main site does not need to be connected
+ *
+ * @param  {Object}    state     Global state tree
+ * @param  {Number}    siteId    The ID of the site we're querying
+ * @return {?Boolean}            Whether site with id equal to siteId is a connected secondary network site
+ */
+export default createSelector(
+	( state, siteId ) => {
+		const isSecondaryNetworkSite = isJetpackSiteSecondaryNetworkSite( state, siteId );
+		if ( isSecondaryNetworkSite === null ) {
+			return null;
+		}
+		if ( ! isSecondaryNetworkSite ) {
+			return false;
+		}
+		const siteIds = Object.keys( get( state, 'sites.items', {} ) );
+		return siteIds.reduce( ( previousResult, mainSiteId ) => {
+			if ( previousResult ) {
+				return previousResult;
```

Comment:
Hi @tyxla, In this case to allow the selector to return null, we don’t want just to return true or false, as lodash some does. It would be possible to use lodash some 2 times, doing 2 iterations one that checks for true and other that checks for null  And if non is true we return false. Using reduce as it is right now we just need one iteration that’s why I used it. The version with lodash some may be easier to understand and If you prefer I can switch.